httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -153,13 +153,19 @@ HttpUriRequest toHttpUriRequest(Request request, Request.Options options) throws
   private ContentType getContentType(Request request) {
     ContentType contentType = ContentType.DEFAULT_TEXT;
     for (Map.Entry<String, Collection<String>> entry : request.headers().entrySet())
-    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
-      Collection values = entry.getValue();
-      if (values != null && !values.isEmpty()) {
-        contentType = ContentType.create(entry.getValue().iterator().next(), request.charset());
-        break;
+      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
+        Collection<String> values = entry.getValue();
+        if (values != null && !values.isEmpty()) {
+          String mimeType = values.iterator().next();
+          try {
+            contentType = ContentType.create(mimeType, request.charset());
+          } catch (IllegalArgumentException e) {
```

Comment:
can you please make a test case that causes the
IllegalArgumentException? copy/pasting stuff works until the
implementation changes invalidating the assumption.

On Wed, Sep 7, 2016 at 1:22 PM, Andrey Stolyarchuk
<notifications@github.com> wrote:
> In httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java:
>
>> @@ -153,13 +153,19 @@ HttpUriRequest toHttpUriRequest(Request request,
>> Request.Options options) throws
>>    private ContentType getContentType(Request request) {
>>      ContentType contentType = ContentType.DEFAULT_TEXT;
>>      for (Map.Entry<String, Collection<String>> entry :
>> request.headers().entrySet())
>> -    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
>> -      Collection values = entry.getValue();
>> -      if (values != null && !values.isEmpty()) {
>> -        contentType =
>> ContentType.create(entry.getValue().iterator().next(), request.charset());
>> -        break;
>> +      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
>> +        Collection<String> values = entry.getValue();
>> +        if (values != null && !values.isEmpty()) {
>> +          String mimeType = values.iterator().next();
>> +          try {
>> +            contentType = ContentType.create(mimeType,
>> request.charset());
>> +          } catch (IllegalArgumentException e) {
>
> Yes, I'm sure it throws IllegalArgumentException in my use case and it broke
> our code, so currently i need to copy-paste this feign class in own code
> base.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.